roles/server/tasks/nginx.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: install nginx
+  apt: name=nginx
+
+- name: create nginx ssl directory
+  file:
+    path=/etc/nginx/ssl
+    state=directory
+
+- name: generate openssl key and crt
+  command: openssl req -new -newkey rsa:4096 -days 3650 -nodes -x509 -subj "/C=CA/ST=British Columbia/L=Vancouver/O=Korobi/OU=Web Development/CN={{ host }}" -keyout /etc/nginx/ssl/{{ site_name }}.key -out /etc/nginx/ssl/{{ site_name }}.crt
+  args:
+    creates: '/etc/nginx/ssl/{{ site_name }}.crt'
```

Comment:
This might work, but I think it's best for projects to have 0 side-effects on the host. Maybe we can provide instructions on how to do it, but I'm not sure it would be great to include a script for it.